### PR TITLE
fix visual regression test

### PR DESCRIPTION
**Pull request by acourdavault**
## Description

visual regression tests were failing:
- viewport size was not enforced
- cypress was running in at most 600 800 resolution in headless ( fixed in 4.1.0 https://github.com/cypress-io/cypress/releases/tag/v4.1.0)
- the assets (screenshots previously recorded included) weere automatically  cleaned

this should be fixed, i updated the error message, cypress to v 4.4.1, the screenshot

## Type of Change
- Bug fix ('fix', 'repair', 'bug')

## Note to Reviewers

the e2e should pass,

if you want you can try to make another screenshot test

### CODE REVIEW

File: packages/manager/cypress/plugins/visualRegPlugin.js [hunk 1]
```diff
@@ -38,6 +38,12 @@ function compareSnapshotsPlugin(args) {
           .pipe(new PNG())
           .on('parsed', function() {
             const imgExpected = this;
+
+            if(imgActual.width != imgExpected.width || imgActual.height != imgExpected.height){
```
Comment (Jskobos) [inline packages/manager/cypress/plugins/visualRegPlugin.js:42]: I'm getting this error consistently.

Comment (acourdavault) [reply packages/manager/cypress/plugins/visualRegPlugin.js:42]: This looks like the fix for cypress version (window size in headless) is not working for you, I am testing this more, thanks

**Comment by acourdavault** [comment]
Finally reproduced it in a clean env.
I enforce the browser to chrome when running in headless mode as Electron headless did not give the proper viewport

**Comment by acourdavault** [comment]
So i had to do multiple changes for this to be useful:
1. use macbook 15 screen size by default to avoid as much as possible a scrollbar on the nav ( in headless mode)
2. record screenshots by defaul tin healdless mode, see `yarn cy:rec-snap`
3. Check that the icon is in the viewport when taking the screenshot
4. NOT testing the "User Setting" button as currently this icon requires the screen to scroll in the viewport, which seems to create a "glitch" in the rendering. The browser has to scroll and we see a duplication of the buffer at the bottom

image

**Comment by acourdavault** [comment]
So a lot of changes on the visual regression:

1. forcing the use of chrome browser both in docker and in headless run to reduce chances of different rendering
2. handling different DPI (retina) displays (scale automatically on comparison the images)
3. Removed the test on "User settings" icon, which being at the bottom seem to force a scroll & re-rendering on the page that does not produce very reproductible result when taking a screenshot


Related to 1) changes the base image of the docker image
Related to 2) switched to Jimp instead of PNGjs to have scaling feature, and change the threashold of the diff to handle variation of quality when resizings (scaling, and different antialiasing mean the same icon will not be 100 prct rendered the same way if the DPI is different)

For more info about issues with viewports/retina see: https://github.com/cypress-io/cypress/issues/1857

File: packages/manager/cypress/integration/smoke-visual-icons.spec.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+          .should('be.visible')
+          .checkSnapshot(s.ariaLabel)
+          .should('be.true');
+      });
+    });
+  });
+  describe('Linodes Landing List Icons', () => {
+    linodeLandingIcons.forEach(i => {
+      return it(`${i.name}`, () => {
+        cy.visitWithLogin('/linodes');
```
Comment (Jskobos) [inline packages/manager/cypress/integration/smoke-visual-icons.spec.js:54]: Maybe you need to repeat the mocking here too? The test above passes but for me this stays on the empty Linode screen so these assertions fail.

Comment (acourdavault) [reply packages/manager/cypress/integration/smoke-visual-icons.spec.js:61]: Thanks @Jskobos good point, did not think of the empty list case

**Comment by Jskobos** [comment]
Sorry this took me a while. The visual test is passing now, but the smoke NB create test is failing consistently, whereas it's fine on develop. The automation seems to get about halfway through the form then stop. Not sure what's going on.

**Comment by acourdavault** [comment]
> Sorry this took me a while. The visual test is passing now, but the smoke NB create test is failing consistently, whereas it's fine on develop. The automation seems to get about halfway through the form then stop. Not sure what's going on.

Thanks, will check this monday. could not this week

**Comment by acourdavault** [comment]
Merged in the change to typescript
